plugins/chat/app/services/chat/mark_all_user_channels_read.rb

```diff
@@ -51,5 +52,15 @@ def mark_associated_mentions_as_read(guardian:, updated_memberships:, **)
         channel_ids: updated_memberships.map(&:channel_id),
       )
     end
+
+    def publish_user_tracking_state(guardian:, updated_memberships:, **)
+      updated_memberships.each do |membership|
+        Chat::Publisher.publish_user_tracking_state(
+          guardian.user,
+          membership.channel_id,
+          membership.last_read_message_id,
+        )
+      end
```

Comment:
That's potentially a lot of MB updates at the same time for one user, I guess that's ok.. the other solution is to be able to send and array of membership updates

Reply:
Yes I was a little worried about that, I think at some point we should add a `chat/bulk-user-tracking` or similar messagebus channel that sends out an object that looks like this for a single user:

```
{
  2332: 667,
  3256: 435
}
```

Where the key is the channel ID and the value is the last read message ID. Might even open a PR for that actually, should be quick